app/BeaconBridge/Config/OpenIdOptions.cs

```diff
@@ -30,6 +30,22 @@ public class OpenIdOptions
   /// </summary>
   ///  TODO in future it should be possible to omit this and use the Client Credentials flow?
   public string Password { get; set; } = string.Empty;
+
+  /// <summary>
+  /// Egress Realm base URL for API interactions with an OpenID Connect (OIDC) compliant
+  /// Identity Provider (IdP), such as Keycloak.
+  /// </summary>
+  public string EgressOpenIdBaseUrl { get; set; } = string.Empty;
+
+  /// <summary>
+  /// Egress Client ID for an OIDC IdP
+  /// </summary>
+  public string EgressClientId { get; set; } = string.Empty;
+
+  /// <summary>
+  /// Egress Client Secret for an OIDC IdP (if the IdP's Client config requires it)
+  /// </summary>
+  public string EgressClientSecret { get; set; } = string.Empty;
```

Comment:
Instead of adding egress specific options to an otherwise generic OpenId options class, look at this: https://learn.microsoft.com/en-us/dotnet/core/extensions/options#named-options-support-using-iconfigurenamedoptions
That way you can configure multiple openid options, reusing the same class

Reply:
It's specific to the submission layer since the `OpenIdBaseUrl` includes the sub realm name. Plus I reuse the username and password for both Egress and Submission layers. I tried the named options but it just over complicates it imo. I could move the egress url client and secret in the EgressOptions instead?

Reply:
Even if you're using the same username and password with submission and egress, you can still have 2 instances of the `OpenIdOptions`. The idea is you're reusing the code. You could have a different realm, username and password. Depending on how many realms/layers you have, you're just endlessly adding those fields to the same options model. But remember those fields are all just variants of a username. password, host, etc.